### PR TITLE
Feature rename board

### PR DESCRIPTION
**Pull request by kadartibi**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -47,6 +47,17 @@ def add_card():
     data_manager.add_new_board('board' + next_board_num)
 
 
+@app.route("/change-board-title", methods=['PUT'])
+@json_response
+def change_board_title():
+    req = request.get_json()
+    print(req)
```
Comment (jsonfly) [inline main.py]: I think we don't need this print here. Btw. the _debugger_ is a more flexible and more useful alternative of _printing things to the console_. :)

File: main.py [hunk 2]
```diff
@@ -41,12 +41,23 @@ def get_statuses():
...
+    new_title = req["newTitle"]
+    board_id = req["boardId"]
+    data_manager.rename_board(board_id, new_title)
+
```
Comment (jsonfly) [inline main.py:61]: When you have nothing to put in the response body, it is recommended to use status code 204, and leave the response body empty. 
(Now we have 'null' as a string in response body, you can see it on the picture attached.)
2019-09-18_15h18_32
Here is a guide how to do it in Flask: [https://www.erol.si/2018/03/flask-return-204-no-content-response/](https://www.erol.si/2018/03/flask-return-204-no-content-response/)

File: static/js/data_handler.js [hunk 3]
```diff
@@ -74,6 +75,46 @@ export let dataHandler = {
     },
     createNewCard: function (cardTitle, boardId, statusId, callback) {
         // creates new card, saves it and calls the callback function with its data
-    }
+    },
     // here comes more features
+    submitNewTitle: function submit_new_title(event) {
```
Comment (jsonfly) [inline static/js/data_handler.js]: Here what you do is:
- define a function named `submit_new_title`
- assign this function to the variable named `dataHandler.submitNewTitle`

This way you will always call this function like `dataHandler.submitNewTitle(...)`, so it is unneccessary to give the initial name `submit_new_title` to it.

TLDR: remove `submit_new_title`

File: static/js/data_handler.js [hunk 4]
```diff
@@ -74,6 +75,46 @@ export let dataHandler = {
...
+                        let header = event.currentTarget.parentElement;
```
Comment (jsonfly) [inline static/js/data_handler.js]: and use ctrl+alt+L (auto-format) on all files to make them consistently formatted

File: static/js/data_handler.js [hunk 5]
```diff
@@ -74,6 +75,46 @@ export let dataHandler = {
...
+                        let newTitle = header.querySelector(".new-board-title");
+                        console.log(newTitle.value);
```
Comment (jsonfly) [inline static/js/data_handler.js]: 🔪 (I always comment a knife (or swords) when I think that a line is unneccessary in the code.) 😄

File: static/js/data_handler.js [hunk 6]
```diff
@@ -74,6 +75,46 @@ export let dataHandler = {
...
+                        if (newTitle.value === ''){
+                            newTitle.value = newTitle.getAttribute("value");
+                        }
+                        let saveButton = header.querySelector(".title-save-button");
+                        header.removeChild(newTitle);
+                        header.removeChild(saveButton);
+                        let boardTitle = document.createElement('span');
```
Comment (jsonfly) [inline static/js/data_handler.js]: When you are not going to write down a line like `boardTitle = something` in the future again, consider using `const` keyword instead of `let`.

File: static/js/data_handler.js [hunk 7]
```diff
@@ -74,6 +75,46 @@ export let dataHandler = {
...
+
+                        let titleAndId = {
+                            newTitle: newTitle.value,
+                            boardId: boardId
+                        };
+                        dom.renameBoard();
+                        fetch(`${window.origin}/change-board-title`, {
+                            method: "PUT",
+                            credentials: "include",
+                            body: JSON.stringify((titleAndId)),
```
Comment (jsonfly) [inline static/js/data_handler.js]: unnecessary duplicated () signs

File: static/js/data_handler.js [hunk 8]
```diff
@@ -74,6 +75,46 @@ export let dataHandler = {
...
+                            cache: "no-cache",
+                            headers: new Headers({
+                                "content-type": "application/json"
+                            })
+                        }).then(function (response) {
+                            if (response.status !== 200){
```
Comment (jsonfly) [inline static/js/data_handler.js]: This logs are useful during debugging, but they should be deleted after we don't need them anymore.

File: static/js/dom.js [hunk 9]
```diff
@@ -122,5 +124,18 @@ export let dom = {
...
         }
+    },
+
+    renameBoard: function () {
+        let boardTitle = document.querySelectorAll('.board-title');
+        for (let title of boardTitle) {
+            title.addEventListener('dblclick', function (event) {
+                let original = event.currentTarget;
+                event.currentTarget.outerHTML = `<input type="text" class="new-board-title" value="${original.textContent}" size="15" required minlength="1" >` +
+                    `<button class="title-save-button">Save</button>`;
```
Comment (jsonfly) [inline static/js/dom.js]: When using the template literals, you can write multiline strings. So you don't need to close the string, write a `+` sign, and start a new string, just hit enter in the string anywhere and continue writing it in the next line.

File: static/js/dom.js [hunk 10]
```diff
@@ -122,5 +124,18 @@ export let dom = {
...
+                let saveButton = document.querySelector('.title-save-button');
+                saveButton.addEventListener('click', dataHandler.submitNewTitle);
```
Comment (jsonfly) [inline static/js/dom.js:146]: Here is the end of my review, I also tried to run the application and I tried the renaming function, it worked well.
Nice work, I had only little things / refactors to mention. Great PR! 😉

File: static/js/data_handler.js [hunk 11]
```diff
@@ -108,10 +108,10 @@ export let dataHandler = {
             })
         }).then(function (response) {
             if (response.status !== 200) {
-                console.log(`RESPONSE STATUS WAS NOT 200: ${response.status}`)
+
```
Comment (jsonfly) [inline static/js/data_handler.js]: This whole .then part can be deleted.
